File: src/javascript/FactorialRecursive.js
/**
 *
 * @param {number} n
 */
function factorial(n) {
    if (n <= 1) {
        return 1;
    }

    return n * factorial(n - 1);
}

function main(){
    console.log('Factorial => ', factorial(3));
}

main();

File: src/ruby/factorial_recursive.rb
# frozen_string_literal: true

def factorial(number)
  aux = 1
  (2..number).each do |x|
    aux *= x
  end
  aux
end

def recursive_factorial(number)
  return 1 if number <= 1

  number * recursive_factorial(number - 1)
end

puts factorial(5)
puts recursive_factorial(5)
